deployment/pypi/setup.py

```diff
@@ -15,6 +17,34 @@
 else:
     raise NotImplementedError('current platform {} not supported'.format(os_type))
 
+class AutoCompletion(install):
+    def run(self):
+        COMP_URL = 'https://raw.githubusercontent.com/microsoft/nni/master/tools/bash-completion'
+        if os_type == 'Linux':
+            HOME = os.environ.get('HOME')
+            if not HOME:
+                install.run(self)
+                return # can not get $HOME, abort.
+            if not os.geteuid(): # run as root
+                BASH_COMP_PREFIX = '/usr/share/bash-completion/completions'
```

Comment:
have you tried macos?

Reply:
sure, it is ok to not support macOS, but what is the consequence of installing nni in macOS? this directory will be created?

Reply:
Only created in Linux. Line 23 do os version check.

Reply:
got it, thanks